src/index.js

```diff
@@ -366,11 +347,19 @@ module.exports = (
       map.sources = map.sources.map(source => {
         // webpack:///webpack:/// happens too for some reason
         while (source.startsWith('webpack:///'))
-          source = source.substr(11);
+          source = source.slice(11);
+        if (source.startsWith('//'))
+          source = source.slice(1);
+        if (source.startsWith('/'))
+          source = relative(process.cwd(), source).replace(/\\/g, '/');
+        if (source.startsWith('external '))
+          source = 'node:' + source.slice(9);
         if (source.startsWith('./'))
-          source = source.substr(2);
+          source = source.slice(2);
+        if (source.startsWith('(webpack)'))
+          source = 'webpack' + source.slice(9);
         if (source.startsWith('webpack/'))
-          return '/webpack/' + source.substr(8);
+          return '/webpack/' + source.slice(8);
         return sourceMapBasePrefix + source;
       });
```

Comment:
This looks really weird. Have you looked into the `output.devtoolModule(Fallback)FilenameTemplate`, which allows to generate sources like you want to.

Reply:
It's really not too much of a problem to have a custom map, but will bear that in mind for future reference, thanks.